supervised/utils/additional_metrics.py

```diff
@@ -137,6 +137,79 @@ def binary_classification(target, predictions, sample_weight=None):
 
         threshold = float(max_metrics["accuracy"]["threshold"])
 
+        accu_threshold_details = {
+            "f1": [],
+            "accuracy": [],
+            "precision": [],
+            "recall": [],
+            "mcc": [],
+        }
+
+        for i in range(STEPS):
```

Comment:
I dont think that loop is needed here. You just want to compute response for threshold that maximize the accuracy metric.

In the code above I've used loop over steps to check different threshold values, here you have the threshold value.

Reply:
Yes will remove the loop and make the changes